Data/Csv.hs

```diff
@@ -124,6 +124,50 @@ import Data.Csv.Types
 --
 -- In practice, the return type of 'decode' rarely needs to be given,
 -- as it can often be inferred from the context.
+-- 
+-- Demonstration of reading from a CSV file/ writing to a CSV file
+-- using the incremental API:
+--
+-- > {-#LANGUAGE DeriveGeneric#-}
+-- > {-#LANGUAGE OverloadedStrings#-}
+-- > {-#LANGUAGE BangPatterns#-}
+-- > 
+-- > import Data.ByteString (ByteString, hGetSome, empty)
+-- > import qualified Data.ByteString.Lazy as BL
+-- > import GHC.Generics
+-- > import Data.Csv.Incremental
+-- > import Data.Csv (FromRecord, ToRecord)    
+-- > import Data.Monoid ((<>), mempty)
+-- > import System.IO
+-- > import System.Exit (exitFailure)
+-- > 
+-- > data Person = Person {
+-- >       name :: ByteString,
```

Comment:
I've made a habbit of making fields strict in the docs when it makes sense, which it does here.